project/make/validate-spaces

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+source "$(dirname "$BASH_SOURCE")/.validate"
+
+#Ignoring files from vendor, ending with .md and .go"
+ignore_files='^vendor/|\.md$|\.go$'
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only | grep -v "$ignore_files" || true) )
+unset IFS
+
+badFiles=()
+for f in "${files[@]}"; do
+    if [ "$(git show "$VALIDATE_HEAD:$f" | grep '[[:space:]]$')" ]; then
+        badFiles+=( "$f" )
+    fi
+done
+
+if [ ${#badFiles[@]} -eq 0 ]; then
+	echo 'Congratulations!  All text files are properly formatted.'
+else
+	{
+		echo "These files have trailing whitespaces:"
+		for f in "${badFiles[@]}"; do
+			echo " - $f"
+		done
+		echo
+		echo 'Please reformat the above files using, for example:'
+		echo '"ex -sc "'"%s/[[:space:]]*$//g|x"'" file" and commit the result.'
```

Comment:
I think that is the reason and from searching on google "Not only does it create a temporary file, it also breaks hard links (eg, instead of changing the contents of the file, it replaces the file with a new file of the same name.) Sometimes this is the desired behavior, sometimes it is acceptable, but when there are multiple links to a file it is almost always the wrong behavior." http://stackoverflow.com/questions/12696125/sed-edit-the-file-in-place